packages/apps/dashboard/server/src/modules/details/details.spec.ts

```diff
@@ -0,0 +1,195 @@
+import { NetworkConfigService } from '../../common/config/network-config.service';
+import { Logger } from '@nestjs/common';
+import { of, throwError } from 'rxjs';
+import {
+  ChainId,
+  ILeader,
+  OperatorUtils,
+  OrderDirection,
+} from '@human-protocol/sdk';
+import { LeadersOrderBy } from '../../common/enums/leader';
+
+jest.mock('@human-protocol/sdk', () => ({
+  ...jest.requireActual('@human-protocol/sdk'),
+  OperatorUtils: {
+    getLeaders: jest.fn(),
+  },
+}));
+
+const MOCK_MAX_LEADERS_COUNT = 5;
+
+jest.mock('../../common/constants/leader', () => ({
+  ...jest.requireActual('../../common/constants/leader'),
+  MAX_LEADERS_COUNT: MOCK_MAX_LEADERS_COUNT,
+}));
+
+describe('DetailsService', () => {
+  let service: DetailsService;
+  let httpService: HttpService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        DetailsService,
+        {
+          provide: HttpService,
+          useValue: {
+            get: jest.fn(),
+          },
+        },
+        {
+          provide: EnvironmentConfigService,
+          useValue: {
+            reputationSource: 'http://reputation.api',
+          },
+        },
+        {
+          provide: NetworkConfigService,
+          useValue: {
+            getAvailableNetworks: jest
+              .fn()
+              .mockResolvedValue([ChainId.MAINNET]),
+          },
+        },
+        Logger,
+      ],
+    }).compile();
+
+    service = module.get<DetailsService>(DetailsService);
+    httpService = module.get<HttpService>(HttpService);
+  });
+
+  it('should fetch and return leaders with reputations', async () => {
+    const mockLeaders = [{ address: '0x123', role: 'Reputation Oracle' }];
+    const mockReputations = [{ address: '0x123', reputation: 'hign' }];
+
+    jest
+      .spyOn(OperatorUtils, 'getLeaders')
+      .mockResolvedValue(mockLeaders as ILeader[]);
+    jest
+      .spyOn(httpService as any, 'get')
+      .mockReturnValue(of({ data: mockReputations }));
+
+    const result = await service.getLeaders(ChainId.ALL);
+
+    expect(result).toEqual([
+      expect.objectContaining({
+        address: '0x123',
+        role: 'Reputation Oracle',
+        reputation: 'hign',
+      }),
+    ]);
+  });
+
+  it('should handle missing reputation data gracefully', async () => {
+    const mockLeaders = [{ address: '0x456', role: 'Job Launcher' }];
+
+    jest
+      .spyOn(OperatorUtils, 'getLeaders')
+      .mockResolvedValue(mockLeaders as ILeader[]);
+    jest.spyOn(httpService as any, 'get').mockReturnValue(of({ data: [] }));
+
+    const result = await service.getLeaders(ChainId.ALL);
+
+    expect(result).toEqual([
+      expect.objectContaining({
+        address: '0x456',
+        role: 'Job Launcher',
+        reputation: 'low',
+      }),
+    ]);
+  });
+
+  it('should return sorted leaders by reputation', async () => {
```

Comment:
This case covered in the next one - `should return first 3 sorted leaders by reputation` - so can be removed